### ISSUE TITLE
W7001 Mapping used in sar location not found

### ISSUE THREAD

**Issue by austinbyers** [aws-cloudformation/cfn-python-lint#1179]
*cfn-lint version: 0.24.6*

Given this CloudFormation template:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Transform: AWS::Serverless-2016-10-31

Mappings:
  Alarms:
    SemanticVersion:
      Value: 1.0.0

Resources:
  MyApplication:
    Type: AWS::Serverless::Application
    Properties:
      Location:
        ApplicationId: arn:aws:serverlessrepo:us-west-1:000000000000:applications/myapp
        SemanticVersion: !FindInMap [Alarms, SemanticVersion, Value]
```

`cfn-lint` gives `W7001 Mapping 'Alarms' is defined but not used`

This is valid CloudFormation - `FindInMap` and `Ref` are (the only) allowed intrinsic functions for the Location of a `Serverless::Application`

**Comment by kddejong** [comment]
I'll take a look into this.  We use the sam translator package to lint the resulting template.  It could be that the translator is removing that Mapping.  We have had to make some changes to handle this for some other things as well.

**Comment by kddejong** [comment]
For tracking purposes this is the translated template.  We will have to check if mappings are used against the pre-transformed template.  I'll get that in.  Thanks for reporting this.
```json
{
  "AWSTemplateFormatVersion": null,
  "Mappings": {
    "Alarms": {
      "SemanticVersion": {
        "Value": "1.0.0"
      }
    }
  },
  "Resources": {
    "MyApplication": {
      "Properties": {
        "Tags": [
          {
            "Key": "lambda:createdBy",
            "Value": "SAM"
          }
        ],
        "TemplateURL": "s3://bucket/value"
      },
      "Type": "AWS::CloudFormation::Stack"
    }
  }
}
```
